File: EventInfoGo/internal/models/fields.go
package models

type Fields struct {
	GlobalActivePower   string `json:"globalActivePower"`
	GlobalReactivePower string `json:"globalReactivePower"`
	Voltage             string `json:"voltage"`
	GlobalIntensity     string `json:"globalIntensity"`
	SubMetering_1       string `json:"subMetering_1"`
	SubMetering_2       string `json:"subMetering_2"`
	SubMetering_3       string `json:"subMetering_3"`
	Timestamp           string `json:"timestamp"`
}

// type Fields struct {
// 	GlobalActivePower   string   `json:"globalActivePower"`
// 	GlobalReactivePower string   `json:"globalReactivePower"`
// 	Voltage             string   `json:"voltage"`
// 	GlobalIntensity     string   `json:"globalIntensity"`
// 	SubMetering_1       string   `json:"subMetering_1"`
// 	SubMetering_2       string   `json:"subMetering_2"`
// 	SubMetering_3       string   `json:"subMetering_3"`
// 	Timestamp           time.Time `json:"timestamp"`
// }

File: EventInfo/DTOs/Fields.cs
namespace EventInfo.DTOs
{
    public class Fields
    {
        public string GlobalActivePower { get; set; } = string.Empty;
        public string GlobalReactivePower { get; set; } = string.Empty;
        public string Voltage { get; set; } = string.Empty;
        public string GlobalIntensity { get; set; } = string.Empty;
        public string SubMetering_1 { get; set; } = string.Empty;
        public string SubMetering_2 { get; set; } = string.Empty;
        public string SubMetering_3 { get; set; } = string.Empty;

        public DateTime Timestamp { get; set; }

    }
}
